WordPress/gutenberg
# Default pattern terminology to Pattern instead of Block Pattern

## What?

This PR replaces mentions of "Block Patterns" with "Patterns" only. 

Fixes #49617.

## Why?

There's an inconsistency in the labeling of the Patterns — sometimes it's "Pattern" while sometimes it's "Block Patterns" — and the goal would be to fix that.

## How?

I mostly searched for the mentions of "Block Patterns" and replaced them manually based on the context.

## Testing Instructions

Basically, a good step would be to search for mentions of patterns in the code and see if makes sense for these contexts.
